### PR TITLE
Add section on how to use with callbacks

### PR DESCRIPTION
**Pull request by ucg8j**
At present, the answer is quite buried in the dash community forum.

### CODE REVIEW

**Comment by chriddyp** [comment]
Thanks @ucg8j ! Could we add two sections here? 
1. Using Callbacks to respond to selections and filters

(fill in with example adapted from `usage.py`)

2. Using Callbacks to populate the table

(fill in with example that has `rows` as an `Output`

***

Also, could we make the examples standalone? Otherwise, we'll inevitably get questions on the forum like "Example doesn't work, DATA is undefined"

**Comment by ucg8j** [comment]
@chriddyp good suggestions. I'll take the datasets from the other examples... I won't be at my computer until tomorrow. But I'll get the changes done then.

**Comment by ucg8j** [comment]
@chriddyp - I've just added a `usage-callback.py`. I _think_ it covers both 1 + 2 as it has a user selection (1) and a callback with `rows` updated in the `Output`. This might not be what you meant though! To clarify, for '1' did you mean responding to user interactions with the table (e.g. selecting rows) which then triggers something else?

**Comment by chriddyp** [comment]
This looks good to me @ucg8j , thanks!

File: README.md [hunk 1]
```diff
@@ -19,5 +19,31 @@ Example from `usage-editable.py`
...
+def update_datatable(user_selection):
+    '''
+    For user selections, return the relevant table
+    '''
+    if user_selection == 'Summary':
+        return DATA.to_dict('records')
+    else:
+        return SOMEOTHERDATA.to_dict('records')
+```
+
```
Comment (chriddyp) [inline README.md]: now that we've added context to this example, would you mind adding a little context to the next two examples that are listed below (`usage.py` and `usage-editable.py`). Something like 
> For usage with row-selection, see `usage.py`. For usage with editable cells, see `usage-editable.py`

**Comment by chriddyp** [comment]
Great, thanks for your patience on this one @ucg8j ! Looks good to me 😄

**Pull request merged**